sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1489,7 +1476,7 @@ class Dataset[T] private[sql](
       case ParserUtils.qualifiedEscapedIdentifier(nameParts, columnNameRegex) =>
         Column(UnresolvedRegex(columnNameRegex, Some(nameParts), caseSensitive))
       case _ =>
-        Column(addDataFrameIdToCol(resolve(colName)))
+        Column(addPlanId(UnresolvedAttribute.quotedString(colName)))
```

Comment:
I think we can still eagerly resolve the column here to make sure it exists, then create an `UnresolvedAttribute` from the resolved `AttributeReference`: `UnresolvedAttribute(Seq(attr.name))`.